module/zfs/vdev.c

```diff
@@ -3454,18 +3464,12 @@ vdev_set_state(vdev_t *vd, boolean_t isopen, vdev_state_t state, vdev_aux_t aux)
 	}
 
 	/*
-	 * Notify ZED of any significant state-change on a leaf vdev.
-	 *
-	 * We ignore transitions from a closed state to healthy unless
-	 * the parent was degraded.
+	 * Notify ZED of any state change to a leaf vdev
 	 */
 	if (vd->vdev_ops->vdev_op_leaf &&
-	    ((save_state > VDEV_STATE_CLOSED) ||
-	    (vd->vdev_state < VDEV_STATE_HEALTHY) ||
-	    (vd->vdev_parent != NULL &&
-	    vd->vdev_parent->vdev_prevstate == VDEV_STATE_DEGRADED))) {
+	    ((save_state != VDEV_STATE_UNKNOWN) &&
+	    (save_state != VDEV_STATE_CLOSED)))
```

Comment:
Yea, I think I mistakenly merged in my older version of this if() block rather than using your newer version.  I'll restore your version.